pkg/api/validate.go

```diff
@@ -27,6 +27,88 @@ var (
 
 	// This regexp is to guard against InvalidDomainNameLabel for hostname validation
 	rxCloudDomainLabel = regexp.MustCompile(`^[a-z][a-z0-9-]{1,61}[a-z0-9]\.`)
+
+	validMasterAndInfraVMSizes = map[VMSize]struct{}{
```

Comment:
The value in the map is a `struct` because this might be more than just a `bool` in the future? Assuming struct is used to store where the size should only be used for testing?

Reply:
map[foo]struct{} is a compact go idiom to indicate a set.  I don't have any immediate intention to add members to the struct.

Reply:
Thanks, understood from standpoint of go idiom. My question was if there is a reason to use a map instead of a list/array of allowed sizes (and I was guessing the purpose using the bool example).